File: 2308-divide-array-into-equal-pairs/divide-array-into-equal-pairs.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) {
        int n = nums.size();
        if (n % 2 != 0) return false;

        vector<int> freq(501, 0);

        for (int num : nums) freq[num]++;

        for (int cnt : freq) {
            if (cnt % 2 != 0) return false;
        }

        return true;
    }
};

File: 2308-divide-array-into-equal-pairs/divide-array-into-equal-pairs.cs
public class Solution {
    public bool DivideArray(int[] nums) {
        int n = nums.Length;
        if (n % 2 != 0) return false;

        int[] freq = new int[501];

        foreach (int num in nums) freq[num]++;

        foreach (int cnt in freq) {
            if (cnt % 2 != 0) return false;
        }

        return true;
    }
}